File: algorithms/codility-master (1)/codility-master/CountingElements/FrogRiverOne/FrogRiverOne.js
function solution(X, A) {
    // write your code in JavaScript (Node.js 8.9.4)
    const result = new Map()
    let resultValue = 0

    if(X === 1 && A[0] === 1) return resultValue

    for (let i = 0; i < A.length; i++) {
        if(!result.has(A[i])) {
            result.set(A[i], i)
        }
    }

    if (!result.has(X) || result.size < X) return -1

    for (let value of result) {
        resultValue = value[1]
    }
    return resultValue
}

File: algorithms/Codility-master/Codility-master/4_CountingElements/FrogRiverOne.py


# 100%
def solution(X, A):
    # write your code in Python 3.6
    nums = [-1] * (X + 1)
    
    for i in range(len(A)):
        if nums[A[i]] ==-1:
            nums[A[i]] = i 
     
    if -1 not in nums[1:]:
        return max(set(nums))

    return -1